src/zenml/integrations/tekton/orchestrators/tekton_orchestrator.py

```diff
@@ -378,106 +420,213 @@ def prepare_or_run_pipeline(
-                for upstream_step_name in step.spec.upstream_steps:
-                    upstream_container_op = step_name_to_container_op[
-                        upstream_step_name
+            Args:
+                yaml_file_path: The path to the YAML file to update.
+                environment: A dictionary of environment variables to add.
+            """
+            pipeline_definition = yaml_utils.read_yaml(pipeline_file_path)
+
+            # Iterate through each component and add the environment variables
+            for executor in pipeline_definition["deploymentSpec"]["executors"]:
+                if (
+                    "container"
+                    in pipeline_definition["deploymentSpec"]["executors"][
+                        executor
                     ]
-                    container_op.after(upstream_container_op)
+                ):
+                    container = pipeline_definition["deploymentSpec"][
+                        "executors"
+                    ][executor]["container"]
+                    if "env" not in container:
+                        container["env"] = []
+                    for key, value in environment.items():
+                        container["env"].append({"name": key, "value": value})
+
+            yaml_utils.write_yaml(pipeline_file_path, pipeline_definition)
+
+            print(
+                f"Updated YAML file with environment variables at {yaml_file_path}"
+            )
+
+        # dynamic_pipeline = _create_dynamic_pipeline()
+
+        # @dsl.pipeline(display_name=orchestrator_run_name)
+        # def _kfp_pipeline() -> None:
+        #     """Create a container_op for each step.
 
-                # Update dictionary of container ops with the current one
-                step_name_to_container_op[step_name] = container_op
+        #     This should contain the name of the docker image and configures the
+        #     entrypoint of the docker image to run the step.
+
+        #     Additionally, this gives each container_op information about its
+        #     direct downstream steps.
+        #     """
+        #     # Dictionary of container_ops index by the associated step name
+        #     step_name_to_container_op: Dict[str, dsl.ContainerOp] = {}
+
+        #     self._configure_pipeline_task(
+        #         pipeline_task=_pipeline_task,
+        #     )
+
+        #     # Find the upstream container ops of the current step and
+        #     # configure the current container op to run after them
+        #     for upstream_step_name in step.spec.upstream_steps:
+        #         upstream_pipeline_task = step_name_to_container_op[
+        #             upstream_step_name
+        #         ]
+        #         _pipeline_task().after(upstream_pipeline_task)
+
+        #     # Update dictionary of container ops with the current one
+        #     step_name_to_container_op[step_name] = _pipeline_task
```

Comment:
```suggestion
```

Reply:
I don't understand this, aren't we doing pipeline already in the `dynamic_pipeline` method

Reply:
Yeah I think this was just code I commented out while I was working on it. So that's why I thought we can just delete it as if you made it work, then it's not needed, right?